platform-metrics/src/main/java/org/hypertrace/core/serviceframework/metrics/PlatformMetricsRegistry.java

```diff
@@ -324,14 +322,27 @@ public static Timer registerTimer(String name, Map<String, String> tags) {
    * default tags also will be reported with the metrics.
    *
    * See https://micrometer.io/docs/concepts#_timers for more details on the Timer.
+   *
+   * Defaults the timer range from 1 second to 60 seconds (both inclusive)
    */
   public static Timer registerTimer(String name, Map<String, String> tags, boolean histogram) {
+    return registerTimer(name, tags, histogram, Duration.ofSeconds(1), Duration.ofSeconds(60));
+  }
+
+  /**
+   * Registers a Timer with the given name with the service's metric registry and reports it
+   * periodically to the configured reporters. Apart from the given tags, the reporting service's
+   * default tags also will be reported with the metrics.
+   *
+   * See https://micrometer.io/docs/concepts#_timers for more details on the Timer.
+   */
+  public static Timer registerTimer(String name, Map<String, String> tags, boolean histogram, Duration minExpectedValue, Duration maxExpectedValue) {
```

Comment:
What is this minExpectedValue and maxExpectedValue

Reply:
For histograms, if these min/max are not configured, then the algorithm will create huge number of buckets (~70 for timers, ~270 for distribution summaries). Each of these buckets translates to 1 time series in prometheus and that causes the prometheus to overflow/OOM/slowdown when we have many such meters.

Hence, micrometer recommends to set these limits.
> minimumExpectedValue/maximumExpectedValue: Controls the number of buckets shipped by publishPercentileHistogram and controls the accuracy and memory footprint of the underlying HdrHistogram structure.


https://micrometer.io/docs/concepts#_histograms_and_percentiles